addons/l10n_in_check_gst_status/models/res_partner.py

```diff
@@ -0,0 +1,108 @@
+        return "https://jva-odoo-iap-apps2-15-0-gst-number-active-check-jva-13181640.dev.odoo.com/iap/l10n_in_reports/1/public/search"
+
+    @api.depends('vat')
+    def _compute_l10n_in_gst_status(self):
+        """
+        Reset GST Status Whenever the `vat` of partner changes
+        """
+        for partner in self:
+            if partner.country_code == 'IN':
+                partner.l10n_in_gstin_verified_status = False
+                partner.l10n_in_gstin_verified_date = False
+
+    @api.depends('l10n_in_gstin_verified_status', 'l10n_in_gstin_verified_date')
+    def _compute_l10n_in_gstin_status_color(self):
+        for partner in self:
+            if partner.country_code == 'IN':
+                partner.l10n_in_gstin_status_color = 'red'
+                if partner.l10n_in_gstin_verified_status == 'active':
+                    diff = (fields.Date.today() - partner.l10n_in_gstin_verified_date).days
+                    if diff >= 0 and diff <= 30:
+                        partner.l10n_in_gstin_status_color = 'green'
+                    elif diff > 30 and diff <= 60:
+                        partner.l10n_in_gstin_status_color = 'yellow'
+            else:
+                partner.l10n_in_gstin_status_color = False
+
+    def l10n_in_get_gstin_verified_status(self):
+        self.ensure_one()
+        url = self._l10n_in_gstin_status_get_url()
+        params = {
+            "account_token": self.env["iap.account"].get("l10n_in_edi").account_token,
+            "dbuuid": self.env["ir.config_parameter"].sudo().get_param("database.uuid"),
+            "gstin_to_search": self.vat,
+        }
+        try:
+            response = jsonrpc(url, params=params, timeout=25)
+        except AccessError:
+            raise UserError(_("Unable to connect"))
+        gst_status = response.get('data', {}).get('sts', "")
+        if gst_status.casefold() == 'active':
+            self.l10n_in_gstin_verified_status = 'active'
+        elif gst_status:
+            self.l10n_in_gstin_verified_status = 'inactive'
+            message = _(
+                 "%s on application of Taxpayer (Effective from %s)",
+                gst_status,
+                response.get('data', {}).get('cxdt', ""),
+            )
+            self.message_post(body=message)
+        elif response.get('error') and any(e.get('code') == 'no-credit' for e in response.get('error')):
+            return self.env["bus.bus"]._sendone(self.env.user.partner_id, "iap_notification",
+                {
+                    "type": "no_credit",
+                    "title": "Not enough credits to check GSTIN status",
+                    "get_credits_url": self.env["iap.account"].get_credits_url(service_name="l10n_in_edi"),
+                },
+            )
+        else:
+            _logger.info("GST status check error %s", response)
+            self.l10n_in_gstin_verified_status = 'invalid'
+        self.write({'l10n_in_gstin_verified_date': fields.Date.today()})
```

Comment:
```suggestion
        self.write({'l10n_in_gstin_verified_status': l10n_in_gstin_verified_status, 'l10n_in_gstin_verified_date': fields.Date.today()})
```
You can use the above suggested variable here
using write at once.

make the addition to a new line with good indentation